lib/ansible/modules/windows/win_iis_webapplication.ps1

```diff
@@ -20,100 +20,82 @@
-If ($name -eq $FALSE) {
-    Fail-Json (New-Object psobject) "missing required argument: name";
-}
-
-# Site
-$site = Get-Attr $params "site" $FALSE;
-If ($site -eq $FALSE) {
-    Fail-Json (New-Object psobject) "missing required argument: site";
-}
+$name = Get-AnsibleParam -obj $params -name "name" -type "str" -failifempty $true
+$site = Get-AnsibleParam -obj $params -name "site" -type "str" -failifempty $true
+$state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "absent","present"
+$physical_path = Get-AnsibleParam -obj $params -name "physical_path" -type "str" -aliases "path"
+$application_pool = Get-AnsibleParam -obj $params -name "application_pool" -type "str"
 
-# State parameter
-$state = Get-Attr $params "state" "present";
-$state.ToString().ToLower();
-If (($state -ne 'present') -and ($state -ne 'absent')) {
-  Fail-Json $result "state is '$state'; must be 'present' or 'absent'"
+$result = @{
+  application_pool = $application_pool
+  changed = $false
+  physical_path = $physical_path
 }
 
-# Path parameter
-$physical_path = Get-Attr $params "physical_path" $FALSE;
-
-# Application Pool Parameter
-$application_pool = Get-Attr $params "application_pool" $FALSE;
-
-
 # Ensure WebAdministration module is loaded
 if ((Get-Module "WebAdministration" -ErrorAction SilentlyContinue) -eq $null) {
   Import-Module WebAdministration
 }
 
-# Result
-$result = New-Object psobject @{
-  application = New-Object psobject
-  changed = $false
-};
-
 # Application info
 $application = Get-WebApplication -Site $site -Name $name
 
 try {
   # Add application
-  If(($state -eq 'present') -and (-not $application)) {
-    If ($physical_path -eq $FALSE) {
-      Fail-Json (New-Object psobject) "missing required arguments: physical_path"
+  if (($state -eq 'present') -and (-not $application)) {
+    if (-not $physical_path) {
+      Fail-Json $result "missing required arguments: path"
     }
-    If (-not (Test-Path $physical_path)) {
-      Fail-Json (New-Object psobject) "specified folder must already exist: physical_path"
+    if (-not (Test-Path -Path $physical_path)) {
+      Fail-Json $result "specified folder must already exist: path"
     }
 
-    $application_parameters = New-Object psobject @{
-      Site = $site
+    $application_parameters = @{
       Name = $name
       PhysicalPath = $physical_path
-    };
+      Site = $site
+    }
 
-    If ($application_pool) {
+    if ($application_pool) {
       $application_parameters.ApplicationPool = $application_pool
     }
 
-    $application = New-WebApplication @application_parameters -Force
+    if (-not $check_mode) {
+        $application = New-WebApplication @application_parameters -Force
+    }
     $result.changed = $true
-
   }
 
   # Remove application
   if ($state -eq 'absent' -and $application) {
-    $application = Remove-WebApplication -Site $site -Name $name
+    $application = Remove-WebApplication -Site $site -Name $name $WhatIf:$check_mode
```

Comment:
BTW, is it OK to correct small typos this way, or should I report an issue and/or make a PR?